examples/ising/ising/model.py

```diff
@@ -0,0 +1,47 @@
+import mesa
+import numpy as np
+
+from .spin import Spin
+
+
+class IsingModel(mesa.Model):
+    def __init__(
+        self,
+        width=50,
+        height=50,
+        spin_up_probability: float = 0.7,
+        temperature: float = 1,
+    ):
+        super().__init__()
+        self.temperature = temperature
+        self.grid = mesa.space.SingleGrid(width, height, torus=True)
+
+        for contents, (x, y) in self.grid.coord_iter():
+            cell = Spin((x, y), self, Spin.DOWN)
+            if self.random.random() < spin_up_probability:
+                cell.state = cell.UP
+            self.grid.place_agent(cell, (x, y))
+
+        self.running = True
+
+    def step(self):
+        agents_list = list(self.agents)
+        self._steps += 1000
+        for i in range(1000):
+            random_spin = self.random.choice(agents_list)
```

Comment:
Having to create a new list of agents from scratch and selecting one is not ideal. I know this is a limitation of `AgentSet`. @EwoutH any idea to do this in a performant way via `AgentSet`? It seems to be a tower of abstractions, that it has slowed down the benchmark for mesa-frames as well: see https://github.com/adamamer20/mesa-frames/issues/25.

Reply:
thanks for the comment. it does not feel right at all haha.

Reply:
Currently I don’t think there is. @vitorfrois could you open a new issue on the main Mesa repo describing this lack of functionality and linking to this PR as example?

Reply:
I was going to suggest you can just do:
```Python
self.agents.select(n=1000)
```
But looking at our current implementation ([here](https://github.com/projectmesa/mesa/blob/50b2964e80897777e92935cb1271cce1c3314845/mesa/agent.py#L124)) that isn’t actually random.

What you can do though is:
```Python
self.agents.shuffle().select(n=1000)
```